ScanExecutorPT fails to run. Throws AccumuloServerException

ScanExecutorPT fails to complete with the following output:

<details open>
<summary>Shell output</summary>

```
Exception in thread "main" java.lang.RuntimeException: java.util.concurrent.ExecutionException: java.lang.RuntimeException: org.apache.accumulo.core.clientImpl.AccumuloServerException: Error on server thor:9997
	at org.apache.accumulo.testing.performance.util.TestExecutor.lambda$stream$0(TestExecutor.java:52)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1655)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:913)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.LongPipeline.collect(LongPipeline.java:491)
	at java.base/java.util.stream.LongPipeline.summaryStatistics(LongPipeline.java:468)
	at org.apache.accumulo.testing.performance.tests.ScanExecutorPT.runShortScans(ScanExecutorPT.java:199)
	at org.apache.accumulo.testing.performance.tests.ScanExecutorPT.runTest(ScanExecutorPT.java:109)
	at org.apache.accumulo.testing.performance.impl.PerfTestRunner.main(PerfTestRunner.java:51)
Caused by: java.util.concurrent.ExecutionException: java.lang.RuntimeException: org.apache.accumulo.core.clientImpl.AccumuloServerException: Error on server thor:9997
	at java.base/java.util.concurrent.FutureTask.report(FutureTask.java:122)
	at java.base/java.util.concurrent.FutureTask.get(FutureTask.java:191)
	at org.apache.accumulo.testing.performance.util.TestExecutor.lambda$stream$0(TestExecutor.java:50)
	... 11 more
Caused by: java.lang.RuntimeException: org.apache.accumulo.core.clientImpl.AccumuloServerException: Error on server thor:9997
	at org.apache.accumulo.core.clientImpl.ScannerIterator.getNextBatch(ScannerIterator.java:185)
	at org.apache.accumulo.core.clientImpl.ScannerIterator.hasNext(ScannerIterator.java:110)
	at com.google.common.collect.Iterators.size(Iterators.java:163)
	at com.google.common.collect.Iterables.size(Iterables.java:126)
	at org.apache.accumulo.testing.performance.tests.ScanExecutorPT.scan(ScanExecutorPT.java:156)
	at org.apache.accumulo.testing.performance.tests.ScanExecutorPT.lambda$runShortScans$1(ScanExecutorPT.java:196)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: org.apache.accumulo.core.clientImpl.AccumuloServerException: Error on server thor:9997
	at org.apache.accumulo.core.clientImpl.ThriftScanner.scan(ThriftScanner.java:324)
	at org.apache.accumulo.core.clientImpl.ScannerIterator.readBatch(ScannerIterator.java:156)
	at org.apache.accumulo.core.clientImpl.ScannerIterator.getNextBatch(ScannerIterator.java:174)
	... 9 more
Caused by: org.apache.thrift.TApplicationException: Internal error processing startScan
	at org.apache.thrift.TServiceClient.receiveBase(TServiceClient.java:79)
	at org.apache.accumulo.core.tabletserver.thrift.TabletClientService$Client.recv_startScan(TabletClientService.java:249)
	at org.apache.accumulo.core.tabletserver.thrift.TabletClientService$Client.startScan(TabletClientService.java:221)
	at org.apache.accumulo.core.clientImpl.ThriftScanner.scan(ThriftScanner.java:453)
	at org.apache.accumulo.core.clientImpl.ThriftScanner.scan(ThriftScanner.java:317)
	... 11 more

```

</details>

<details open>
<summary>Server logs</summary>

```
java.lang.IllegalArgumentException: Invalid option heed_hints at 
org.apache.accumulo.core.spi.scan.SimpleScanDispatcher.lambda$init$0(SimpleScanDispatcher.java:105) at 
com.google.common.collect.RegularImmutableMap.forEach(RegularImmutableMap.java:185) at 
org.apache.accumulo.core.spi.scan.SimpleScanDispatcher.init(SimpleScanDispatcher.java:81) at 
org.apache.accumulo.server.conf.TableConfiguration.createScanDispatcher(TableConfiguration.java:215) at 
org.apache.accumulo.server.conf.TableConfiguration.lambda$new$1(TableConfiguration.java:82) at 
org.apache.accumulo.core.conf.AccumuloConfiguration$DeriverImpl.derive(AccumuloConfiguration.java:482) at 
org.apache.accumulo.server.conf.TableConfiguration.getScanDispatcher(TableConfiguration.java:270) at 
org.apache.accumulo.tserver.ThriftClientHandler.getScanDispatcher(ThriftClientHandler.java:272) at 
org.apache.accumulo.tserver.ThriftClientHandler.continueScan(ThriftClientHandler.java:378) at 
org.apache.accumulo.tserver.ThriftClientHandler.startScan(ThriftClientHandler.java:342) at 
jdk.internal.reflect.GeneratedMethodAccessor4.invoke(Unknown Source) at 
java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) at 
java.base/java.lang.reflect.Method.invoke(Method.java:566) at 
org.apache.accumulo.core.trace.TraceUtil.lambda$wrapService$6(TraceUtil.java:235) at 
com.sun.proxy.$Proxy38.startScan(Unknown Source) at 
org.apache.accumulo.core.tabletserver.thrift.TabletClientService$Processor$startScan.getResult(TabletClientService.java:2944) at 
org.apache.accumulo.core.tabletserver.thrift.TabletClientService$Processor$startScan.getResult(TabletClientService.java:2923)
 at org.apache.thrift.ProcessFunction.process(ProcessFunction.java:38) at 
org.apache.thrift.TBaseProcessor.process(TBaseProcessor.java:38) at 
org.apache.accumulo.server.rpc.TimedProcessor.process(TimedProcessor.java:63) at 
org.apache.thrift.server.AbstractNonblockingServer$FrameBuffer.invoke(AbstractNonblockingServer.java:524) at 
org.apache.accumulo.server.rpc.CustomNonBlockingServer$CustomFrameBuffer.invoke(CustomNonBlockingServer.java:114) at
 org.apache.thrift.server.Invocation.run(Invocation.java:18) at 
java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) at 
java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) at 
java.base/java.lang.Thread.run(Thread.java:829) 
```

</details>

--- a/src/main/java/org/apache/accumulo/testing/performance/tests/ScanExecutorPT.java
+++ b/src/main/java/org/apache/accumulo/testing/performance/tests/ScanExecutorPT.java
@@ -57,10 +57,10 @@ public class ScanExecutorPT implements PerformanceTest {
   private static final String SCAN_EXECUTOR_THREADS = "2";
   private static final String SCAN_PRIORITIZER = "org.apache.accumulo.core.spi.scan.HintScanPrioritizer";
 
-  private static final String TEST_DESC = "Scan Executor Test.  Test running lots of short scans "
-      + "while long scans are running in the background.  Each short scan reads a random row and "
+  private static final String TEST_DESC = "Scan Executor Test. Test running lots of short scans "
+      + "while long scans are running in the background. Each short scan reads a random row and "
       + "family. Using execution hints, short scans are randomly either given a high priority or "
-      + "a dedicated executor.  If the scan prioritizer or dispatcher is not working properly, "
+      + "a dedicated executor. If the scan prioritizer or dispatcher is not working properly, "
       + "then the short scans will be orders of magnitude slower.";
 
   @Override
@@ -122,21 +122,21 @@ public class ScanExecutorPT implements PerformanceTest {
     final String ms = TimeUnit.MILLISECONDS.toString();
 
     builder.id("sexec").description(TEST_DESC);
-    builder.info("write", NUM_ROWS * NUM_FAMS * NUM_QUALS, t2 - t1, "entries/sec",
+    builder.info("write_rate", NUM_ROWS * NUM_FAMS * NUM_QUALS, t2 - t1, "entries/sec",
         "Data write rate");
-    builder.info("compact", NUM_ROWS * NUM_FAMS * NUM_QUALS, t3 - t2, "entries/sec",
+    builder.info("compact_rate", NUM_ROWS * NUM_FAMS * NUM_QUALS, t3 - t2, "entries/sec",
         "Compact rate");
     builder.info("short_times1", shortStats1, ms, "Duration of each short scan from first run.");
     builder.info("short_times2", shortStats2, ms, "Duration of each short scan from second run.");
-    builder.result("short", shortStats2.getAverage(), ms,
+    builder.result("short_average", shortStats2.getAverage(), ms,
         "Average duration of short scans from second run.");
     builder.info("long_counts", longStats, "entries read",
         "Entries read by each long scan threads");
-    builder.info("long", longStats.getSum(), (t4 - t3), "entries/sec",
+    builder.info("long_rate", longStats.getSum(), (t4 - t3), "entries/sec",
         "Combined rate of all long scans");
     builder.parameter("short_threads", NUM_SHORT_SCANS_THREADS, "Threads used to run short scans.");
     builder.parameter("long_threads", NUM_LONG_SCANS,
-        "Threads running long scans.  Each thread repeatedly scans entire table for duration of test.");
+        "Threads running long scans. Each thread repeatedly scans entire table for duration of test.");
     builder.parameter("rows", NUM_ROWS, "Rows in test table");
     builder.parameter("families", NUM_FAMS, "Families per row in test table");
     builder.parameter("qualifiers", NUM_QUALS, "Qualifiers per family in test table");

--- a/src/main/java/org/apache/accumulo/testing/performance/tests/ScanExecutorPT.java
+++ b/src/main/java/org/apache/accumulo/testing/performance/tests/ScanExecutorPT.java
@@ -23,7 +23,6 @@ import java.util.LongSummaryStatistics;
 import java.util.Map;
 import java.util.Map.Entry;
 import java.util.Random;
-import java.util.concurrent.ExecutionException;
 import java.util.concurrent.TimeUnit;
 import java.util.concurrent.atomic.AtomicBoolean;
 
@@ -75,10 +74,17 @@ public class ScanExecutorPT implements PerformanceTest {
         SCAN_EXECUTOR_THREADS);
     siteCfg.put(Property.TSERV_SCAN_EXECUTORS_PREFIX.getKey() + "se1.prioritizer",
         SCAN_PRIORITIZER);
+    siteCfg.put(
+        Property.TSERV_SCAN_EXECUTORS_PREFIX.getKey() + "se1.prioritizer.opts.priority.se1p1", "1");
+    siteCfg.put(
+        Property.TSERV_SCAN_EXECUTORS_PREFIX.getKey() + "se1.prioritizer.opts.priority.se1p2", "2");
+
     siteCfg.put(Property.TSERV_SCAN_EXECUTORS_PREFIX.getKey() + "se2.threads",
         SCAN_EXECUTOR_THREADS);
     siteCfg.put(Property.TSERV_SCAN_EXECUTORS_PREFIX.getKey() + "se2.prioritizer",
         SCAN_PRIORITIZER);
+    siteCfg.put(
+        Property.TSERV_SCAN_EXECUTORS_PREFIX.getKey() + "se2.prioritizer.opts.priority.se2p1", "1");
 
     return new SystemConfiguration().setAccumuloConfig(siteCfg);
   }
@@ -89,8 +95,9 @@ public class ScanExecutorPT implements PerformanceTest {
     String tableName = "scept";
 
     Map<String,String> props = new HashMap<>();
-    props.put(Property.TABLE_SCAN_DISPATCHER_OPTS.getKey() + "executor", "se1");
-    props.put(Property.TABLE_SCAN_DISPATCHER_OPTS.getKey() + "heed_hints", "true");
+    props.put(Property.TABLE_SCAN_DISPATCHER_OPTS.getKey() + "executor.se1p1", "se1");
+    props.put(Property.TABLE_SCAN_DISPATCHER_OPTS.getKey() + "executor.se1p2", "se1");
+    props.put(Property.TABLE_SCAN_DISPATCHER_OPTS.getKey() + "executor.se2p1", "se2");
     props.put(Property.TABLE_BLOCKCACHE_ENABLED.getKey(), "true");
 
     env.getClient().tableOperations().create(tableName,
@@ -178,11 +185,10 @@ public class ScanExecutorPT implements PerformanceTest {
     return count;
   }
 
-  private LongSummaryStatistics runShortScans(Environment env, String tableName, int numScans)
-      throws InterruptedException, ExecutionException {
+  private LongSummaryStatistics runShortScans(Environment env, String tableName, int numScans) {
 
-    Map<String,String> execHints = ImmutableMap.of("executor", "se2");
-    Map<String,String> prioHints = ImmutableMap.of("priority", "1");
+    Map<String,String> execHints = ImmutableMap.of("scan_type", "se2p1");
+    Map<String,String> prioHints = ImmutableMap.of("scan_type", "se1p1");
 
     try (TestExecutor<Long> executor = new TestExecutor<>(NUM_SHORT_SCANS_THREADS)) {
       Random rand = new Random();
@@ -201,7 +207,7 @@ public class ScanExecutorPT implements PerformanceTest {
   }
 
   private TestExecutor<Long> startLongScans(Environment env, String tableName, AtomicBoolean stop) {
-    Map<String,String> hints = ImmutableMap.of("priority", "2");
+    Map<String,String> hints = Map.of("scan_type", "se1p2");
 
     TestExecutor<Long> longScans = new TestExecutor<>(NUM_LONG_SCANS);
 

--- a/src/main/java/org/apache/accumulo/testing/performance/tests/ScanExecutorPT.java
+++ b/src/main/java/org/apache/accumulo/testing/performance/tests/ScanExecutorPT.java
@@ -75,16 +75,14 @@ public class ScanExecutorPT implements PerformanceTest {
     siteCfg.put(Property.TSERV_SCAN_EXECUTORS_PREFIX.getKey() + "se1.prioritizer",
         SCAN_PRIORITIZER);
     siteCfg.put(
-        Property.TSERV_SCAN_EXECUTORS_PREFIX.getKey() + "se1.prioritizer.opts.priority.se1p1", "1");
+        Property.TSERV_SCAN_EXECUTORS_PREFIX.getKey() + "se1.prioritizer.opts.priority.short", "1");
     siteCfg.put(
-        Property.TSERV_SCAN_EXECUTORS_PREFIX.getKey() + "se1.prioritizer.opts.priority.se1p2", "2");
+        Property.TSERV_SCAN_EXECUTORS_PREFIX.getKey() + "se1.prioritizer.opts.priority.long", "2");
 
     siteCfg.put(Property.TSERV_SCAN_EXECUTORS_PREFIX.getKey() + "se2.threads",
         SCAN_EXECUTOR_THREADS);
     siteCfg.put(Property.TSERV_SCAN_EXECUTORS_PREFIX.getKey() + "se2.prioritizer",
         SCAN_PRIORITIZER);
-    siteCfg.put(
-        Property.TSERV_SCAN_EXECUTORS_PREFIX.getKey() + "se2.prioritizer.opts.priority.se2p1", "1");
 
     return new SystemConfiguration().setAccumuloConfig(siteCfg);
   }
@@ -95,9 +93,8 @@ public class ScanExecutorPT implements PerformanceTest {
     String tableName = "scept";
 
     Map<String,String> props = new HashMap<>();
-    props.put(Property.TABLE_SCAN_DISPATCHER_OPTS.getKey() + "executor.se1p1", "se1");
-    props.put(Property.TABLE_SCAN_DISPATCHER_OPTS.getKey() + "executor.se1p2", "se1");
-    props.put(Property.TABLE_SCAN_DISPATCHER_OPTS.getKey() + "executor.se2p1", "se2");
+    props.put(Property.TABLE_SCAN_DISPATCHER_OPTS.getKey() + "executor", "se1");
+    props.put(Property.TABLE_SCAN_DISPATCHER_OPTS.getKey() + "executor.dedicated", "se2");
     props.put(Property.TABLE_BLOCKCACHE_ENABLED.getKey(), "true");
 
     env.getClient().tableOperations().create(tableName,
@@ -187,8 +184,8 @@ public class ScanExecutorPT implements PerformanceTest {
 
   private LongSummaryStatistics runShortScans(Environment env, String tableName, int numScans) {
 
-    Map<String,String> execHints = ImmutableMap.of("scan_type", "se2p1");
-    Map<String,String> prioHints = ImmutableMap.of("scan_type", "se1p1");
+    Map<String,String> dHints = ImmutableMap.of("scan_type", "dedicated");
+    Map<String,String> sHints = ImmutableMap.of("scan_type", "short");
 
     try (TestExecutor<Long> executor = new TestExecutor<>(NUM_SHORT_SCANS_THREADS)) {
       Random rand = new Random();
@@ -207,7 +204,7 @@ public class ScanExecutorPT implements PerformanceTest {
   }
 
   private TestExecutor<Long> startLongScans(Environment env, String tableName, AtomicBoolean stop) {
-    Map<String,String> hints = Map.of("scan_type", "se1p2");
+    Map<String,String> hints = Map.of("scan_type", "long");
 
     TestExecutor<Long> longScans = new TestExecutor<>(NUM_LONG_SCANS);
 

--- a/src/main/java/org/apache/accumulo/testing/performance/tests/ScanExecutorPT.java
+++ b/src/main/java/org/apache/accumulo/testing/performance/tests/ScanExecutorPT.java
@@ -43,7 +43,6 @@ import org.apache.accumulo.testing.performance.util.TestData;
 import org.apache.accumulo.testing.performance.util.TestExecutor;
 import org.apache.hadoop.io.Text;
 
-import com.google.common.collect.ImmutableMap;
 import com.google.common.collect.Iterables;
 
 public class ScanExecutorPT implements PerformanceTest {
@@ -70,19 +69,16 @@ public class ScanExecutorPT implements PerformanceTest {
 
     siteCfg.put(Property.TSERV_SCAN_MAX_OPENFILES.getKey(), "200");
     siteCfg.put(Property.TSERV_MINTHREADS.getKey(), "200");
-    siteCfg.put(Property.TSERV_SCAN_EXECUTORS_PREFIX.getKey() + "se1.threads",
-        SCAN_EXECUTOR_THREADS);
-    siteCfg.put(Property.TSERV_SCAN_EXECUTORS_PREFIX.getKey() + "se1.prioritizer",
-        SCAN_PRIORITIZER);
-    siteCfg.put(
-        Property.TSERV_SCAN_EXECUTORS_PREFIX.getKey() + "se1.prioritizer.opts.priority.short", "1");
-    siteCfg.put(
-        Property.TSERV_SCAN_EXECUTORS_PREFIX.getKey() + "se1.prioritizer.opts.priority.long", "2");
 
-    siteCfg.put(Property.TSERV_SCAN_EXECUTORS_PREFIX.getKey() + "se2.threads",
-        SCAN_EXECUTOR_THREADS);
-    siteCfg.put(Property.TSERV_SCAN_EXECUTORS_PREFIX.getKey() + "se2.prioritizer",
-        SCAN_PRIORITIZER);
+    final String tservSEprefix = Property.TSERV_SCAN_EXECUTORS_PREFIX.getKey();
+
+    siteCfg.put(tservSEprefix + "se1.threads", SCAN_EXECUTOR_THREADS);
+    siteCfg.put(tservSEprefix + "se1.prioritizer", SCAN_PRIORITIZER);
+    siteCfg.put(tservSEprefix + "se1.prioritizer.opts.priority.short", "1");
+    siteCfg.put(tservSEprefix + "se1.prioritizer.opts.priority.long", "2");
+
+    siteCfg.put(tservSEprefix + "se2.threads", SCAN_EXECUTOR_THREADS);
+    siteCfg.put(tservSEprefix + "se2.prioritizer", SCAN_PRIORITIZER);
 
     return new SystemConfiguration().setAccumuloConfig(siteCfg);
   }
@@ -90,28 +86,29 @@ public class ScanExecutorPT implements PerformanceTest {
   @Override
   public Report runTest(Environment env) throws Exception {
 
-    String tableName = "scept";
+    AccumuloClient client = env.getClient();
+
+    final String tableName = "scept";
 
     Map<String,String> props = new HashMap<>();
     props.put(Property.TABLE_SCAN_DISPATCHER_OPTS.getKey() + "executor", "se1");
     props.put(Property.TABLE_SCAN_DISPATCHER_OPTS.getKey() + "executor.dedicated", "se2");
     props.put(Property.TABLE_BLOCKCACHE_ENABLED.getKey(), "true");
 
-    env.getClient().tableOperations().create(tableName,
-        new NewTableConfiguration().setProperties(props));
+    client.tableOperations().create(tableName, new NewTableConfiguration().setProperties(props));
 
     long t1 = System.currentTimeMillis();
-    TestData.generate(env.getClient(), tableName, NUM_ROWS, NUM_FAMS, NUM_QUALS);
+    TestData.generate(client, tableName, NUM_ROWS, NUM_FAMS, NUM_QUALS);
     long t2 = System.currentTimeMillis();
-    env.getClient().tableOperations().compact(tableName, null, null, true, true);
+    client.tableOperations().compact(tableName, null, null, true, true);
     long t3 = System.currentTimeMillis();
 
     AtomicBoolean stop = new AtomicBoolean(false);
 
-    TestExecutor<Long> longScans = startLongScans(env, tableName, stop);
+    TestExecutor<Long> longScans = startLongScans(client, tableName, stop);
 
-    LongSummaryStatistics shortStats1 = runShortScans(env, tableName, 50000);
-    LongSummaryStatistics shortStats2 = runShortScans(env, tableName, 100000);
+    LongSummaryStatistics shortStats1 = runShortScans(client, tableName, 50000);
+    LongSummaryStatistics shortStats2 = runShortScans(client, tableName, 100000);
 
     stop.set(true);
     long t4 = System.currentTimeMillis();
@@ -182,10 +179,11 @@ public class ScanExecutorPT implements PerformanceTest {
     return count;
   }
 
-  private LongSummaryStatistics runShortScans(Environment env, String tableName, int numScans) {
+  private LongSummaryStatistics runShortScans(AccumuloClient client, String tableName,
+      int numScans) {
 
-    Map<String,String> dHints = ImmutableMap.of("scan_type", "dedicated");
-    Map<String,String> sHints = ImmutableMap.of("scan_type", "short");
+    Map<String,String> execHints = Map.of("scan_type", "dedicated");
+    Map<String,String> prioHints = Map.of("scan_type", "short");
 
     try (TestExecutor<Long> executor = new TestExecutor<>(NUM_SHORT_SCANS_THREADS)) {
       Random rand = new Random();
@@ -196,20 +194,21 @@ public class ScanExecutorPT implements PerformanceTest {
         // scans have a 20% chance of getting dedicated thread pool and 80% chance of getting high
         // priority
         Map<String,String> hints = rand.nextInt(10) <= 1 ? execHints : prioHints;
-        executor.submit(() -> scan(tableName, env.getClient(), row, fam, hints));
+        executor.submit(() -> scan(tableName, client, row, fam, hints));
       }
 
       return executor.stream().mapToLong(l -> l).summaryStatistics();
     }
   }
 
-  private TestExecutor<Long> startLongScans(Environment env, String tableName, AtomicBoolean stop) {
+  private TestExecutor<Long> startLongScans(AccumuloClient client, String tableName,
+      AtomicBoolean stop) {
     Map<String,String> hints = Map.of("scan_type", "long");
 
     TestExecutor<Long> longScans = new TestExecutor<>(NUM_LONG_SCANS);
 
     for (int i = 0; i < NUM_LONG_SCANS; i++) {
-      longScans.submit(() -> scan(tableName, env.getClient(), stop, hints));
+      longScans.submit(() -> scan(tableName, client, stop, hints));
     }
     return longScans;
   }

